models/register.php

```diff
@@ -2,13 +2,127 @@
+        if (isset($_POST["insert"])){
+            // 入力チェック
+            $input_check = new Account();
+            $username_message = $input_check->Get_account_name_errorMessage();
+            $email_message = $input_check->Get_email_errorMessage();
+            $password_message = $input_check->Get_password_errorMessage();
+            if($username_message || $email_message || $password_message){
+                $this->page_data['username_errorMessage'] = $username_message;
+                $this->page_data['email_errorMessage'] = $email_message;
+                $this->page_data['password_errorMessage'] = $password_message;
+            }else{
+                $input_check->Insert_Accounts();
+                $errormessage = $input_check->Get_errorMessage();
+                if($errormessage){
+                    echo "<script>alert('$errormessage');</script>";
+                }
+            }
+        }
+    }
+
+}
+
+class Account {
+    private $account_name;              // ユーザ名
+    private $email;                     // メールアドレス
+    private $password;                  // パスワード
+    private $errorflg;                  // エラーフラグ
+    private $errorMessage;              // 汎用エラー
+    private $account_name_errorMessage; // ユーザ名エラーメッセージ
+    private $email_errorMessage;        // メールアドレスエラーメッセージ
+    private $password_errorMessage;     // パスワードエラーメッセージ
+
+    function __construct(){
+        // 画面から送信された内容を取得
+        $this->account_name = $_POST['account_name'];
+        if(!isset($this->account_name)){
+            $this->account_name = '';
+        }
+        $this->email = $_POST['email'];
+        if(!isset($this->email)){
+            $this->email = '';
+        }
+        $this->password = $_POST['password'];
+        if(!isset($this->password)){
+            $this->password = '';
+        }
+        $this->Input_check();
+    }
+
+    function Insert_Accounts(){
+        try{
+            //データベース接続
+            $DB = new DB_Connect();
+            $pdo = $DB->getPDO();
+            // SQL準備
+            $stmt = $pdo->prepare('INSERT INTO Accounts (account_name, email, password_hash)VALUES(:account_name, :email, :password_hash)');
+            // プレースホルダの値をセット
+            $password_hash = password_hash($this->password, PASSWORD_BCRYPT);
+            $stmt->bindParam(':account_name',$this->account_name, PDO::PARAM_STR);
+            $stmt->bindParam(':email', $this->email, PDO::PARAM_STR);
+            $stmt->bindParam(':password_hash', $password_hash, PDO::PARAM_STR);
+            // ＳＱＬ実行部分
+            $check = $stmt->execute();
+            if($check){
+                echo "<script>alert('アカウントの登録に成功しました！');</script>";
+            }else{
+                echo "<script>alert('アカウントの登録に失敗しました。');</script>";
+            }
+        }catch(PDOException $e){
+            $this->errorMessage .= 'データベース更新エラー\n'.addslashes($e->getMessage()).'。';
+        }
+    }
+
+    protected function Input_check() {

```

Comment:
コメントぉ…(´・ω・｀)

protectedは、「このclassを継承した子classでも使用できます」という宣言です。
このclassがそもそも継承させる事を目的としない場合は、protected宣言は不要です。

Input_check()はこのclass内だけで使用しているみたいなので、
privateという宣言にすると、外部からは参照できなくなります。

```suggestion
    private function Input_check() {
```